plugin/plugin_read.go

```diff
@@ -0,0 +1,44 @@
+package plugin
+
+import (
+	"context"
+	"sort"
+
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/apache/arrow/go/v13/arrow/array"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+)
+
+// readAll is used in tests to read all records from a table.
+func (p *Plugin) readAll(ctx context.Context, table *schema.Table) ([]arrow.Record, error) {
+	// TODO (Go 1.21+) use testing.Testing() to guard against use of this function outside of tests
+	// https://stackoverflow.com/a/75787351
+
+	var err error
+	ch := make(chan arrow.Record)
+	go func() {
+		defer close(ch)
+		err = p.client.Read(ctx, table, ch)
+	}()
+	// nolint:prealloc
+	var records []arrow.Record
+	for record := range ch {
+		records = append(records, record)
+	}
+
+	// sort records by "id" column, if present. Because "id" is auto-incrementing in the test
+	// data generator, this should result in records being returned in insertion order.
+	sch := table.ToArrowSchema()
+	if sch.HasField("id") {
```

Comment:
I think this should be done outside of `read` with a helper function because we might want to sort it with different logic and then it will either be impossible or this function would grow.

Reply:
The vanilla read should just return it as it is as it is not aware of what columns are there and so on.

Reply:
This is a private function, I figured we can refactor it later if/when the need arises. But okay, I can do it now

Reply:
Refactored 👍 I also added some random shuffling to the memdb Read function so that we can be sure this is used everywhere it needs to be used